Cargo.toml

```diff
@@ -148,7 +148,7 @@ dotenvy = { version = "0.15.0", default-features = false }
 
 # Runtimes
 [workspace.dependencies.async-std]
-version = "1.12"
```

Comment:
Strictly speaking, this version doesn't need to be advanced. 

In fact, doing so could be undesirable, because if someone has the dependency pinned for whatever reason,
this will result in them having both `1.12` and `1.13` in their project.

Reply:
lol @paolobarbolini you're fast

Reply:
Strictly speaking no.

However it is one of many approaches.
1. Keep testing the sqlx project using the locked 1.12
2. Update the lock here, so sqlx project is testing using the locked 1.13, but allow users to keep using 1.12 even though it isnt tested by the sqlx project
3. Update the toml, so users are using the same version tested here.

Another harder one is to test sqlx using multiple versions of async-std.

IMO this is a big enough leap forward it is worth going with option 3.
End-users are going to be running `cargo update -p async-std` , so they will be using v1.13, so option 1 is a bad idea, and option 2 isnt great either.

Regarding users potentially having two versions of async-std in their project, they can be using cargo-deny to track and prevent that, and then make a decision whether to upgrade sqlx.

Anyways, I am happy to revert this bit if you prefer option 2.

If you prefer Option 1, close the PR.

Reply:
We can't assume what users are going to do. Wholesale bumps have gotten us in trouble in the past, so I'm not inclined to merge this as-is anytime before 0.9.0.

Option 2 is not unreasonable because while we won't _test_ with anything older, we do at least check that it builds: https://github.com/launchbadge/sqlx/blob/10bec327703303eca5f8ec5809c4713a4294313d/.github/workflows/sqlx.yml#L53-L61

Reply:
Removed as requested.